pkg/jobs/status.go

```diff
@@ -2,17 +2,30 @@ package jobs
 
 import (
 	"context"
+	"fmt"
 	"time"
 
 	"github.com/deepfence/YaraHunter/pkg/output"
+	"github.com/deepfence/YaraHunter/pkg/scan"
+	log "github.com/sirupsen/logrus"
 )
 
-func StartStatusReporter(ctx context.Context, scan_id string) chan error {
+func StartStatusReporter(ctx context.Context, scan_id string, scanner *scan.Scanner) chan error {
+
 	res := make(chan error)
 	startScanJob()
+
+	//If we don't get any active status back within threshold,
+	//we consider the scan job as dead
+	threshold := *scanner.InactiveThreshold
+
 	go func() {
-		defer stopScanJob()
+		defer func() {
+			stopScanJob()
+		}()
```

Comment:
What benefit over `defer stopScanJob()`?

Reply:
This is a leftover... I initially added abort/exit flag in the defer... but later moved that out. 
I will revert this.